src/store/actions/filter.action.ts

```diff
@@ -0,0 +1,6 @@
+import { createCustomAction } from 'typesafe-actions';
+import { Data } from '../../pages/comparison/interfaces/filter.interface';
+
+export const filterAction = createCustomAction('FILTER', (data: Data | null) => {
```

Comment:
`Data | null`? Or just `Data`?

Reply:
Yes, exactly Data or null,  
because null i used for "clean"  filter object, this is to make it easier to check
if (null) {do nothing}, else {sort array...}

Reply:
Understandable. I think that TypeScript allows to use just `Data` and it includes under the hud that null can be passed. I mean, if you remove `| null` check it will works anyway. But if you want to specify it explicitly it also okay.

Reply:
of course i tried to remove "| null" , but got the error ...

Reply:
Error:
Argument of type 'null' is not assignable to parameter of type 'Data'.ts(2345)